config/locales/en.yml

```diff
@@ -150,7 +150,11 @@ en:
       intermediate_rent_product_name:
         blank: "Enter name of other intermediate rent product"
       saledate:
-        financial_year: "Date must be from 22/23 financial year, which is between 1st April 2022 and 31st March 2023"
+        current_financial_year:
+          Enter a date within the %{current_start_year_short}/%{current_end_year_short} financial year, which is between 1st April %{current_start_year_long} and 31st March %{current_end_year_long}
+        previous_and_current_financial_year:
+          "Enter a date within the %{previous_start_year_short}/%{previous_end_year_short} or %{previous_end_year_short}/%{current_end_year_short} financial years, which is between 1st April %{previous_start_year_long} and 31st March %{current_end_year_long}"
+
```

Comment:
would it make sense to pass in those dates fully as opposed to hardcode April 1st and March 31st in the error message?